pycbc/inject/inject.py

```diff
@@ -568,6 +568,7 @@ def apply(self, strain, detector_name, f_lower=None, distance_scale=1,
             # safety buffer
             start_time = inj.tc - 2 * (inj_length + 1)
             if end_time < t0 or start_time > t1:
+                logger.warning("No injections applied.")
```

Comment:
Shouldn't this be after the for loop?

Reply:
See discussion here https://book.pythontips.com/en/latest/for_-_else.html

Reply:
After which loop? If there are no injections, `if end_time < t0 or start_time > t1:` will be True and the rest of the interaction will be skipped, right? That's why I put the warning in the loop.

Reply:
That's only true for a *specific* injection. What if the file has multiple things which could be injected? The injection file can have many signals in it (perhaps for different times or even overlapping).